test/Application/ConfigInjectionTest.php

```diff
@@ -99,6 +134,44 @@ public static function assertPipelineContainsInstanceOf($class, $pipeline, $mess
         Assert::assertThat($found, Assert::isTrue(), $message);
     }
 
+    public function assertRouteMiddleware(MiddlewareInterface $middleware)
+    {
+        if ($middleware instanceof Middleware\RouteMiddleware) {
+            Assert::that($middleware instanceof Middleware\RouteMiddleware);
+            return;
+        }
+
+        if (! $middleware instanceof Middleware\LazyLoadingMiddleware) {
+            Assert::fail('Middleware is not an instance of RouteMiddleware');
+        }
+
+        Assert::assertAttributeSame(
```

Comment:
Why `Assert::` not `self::` or `$this->` ?

Reply:
We cannot use `$this->` in static context (thanks for pointing out that change!).

Otherwise, `Assert::` is consistent with the existing custom assertion methods, and what the PHPUnit manual suggests for custom assertions.